app/src/main/java/org/mozilla/focus/Components.kt

```diff
@@ -126,6 +128,10 @@ class Components(
         }
     }
 
+    val browserScreenStore by lazy {
```

Comment:
This shows an issue with our first proposal of installing a Middleware to BrowserStore - which is a singleton per app so if the Middleware is to dispatch updates to another store like a `..ScreenStore` that also has to be a singleton and this diverges from our flow of having each store's lifecycle tied to it's component's - most likely a fragment - lifecycle.
Thinking about how this will evolve over time, we'd have multiple `..ScreenStore` singletons, if not set as private here accessible from any other component. This could bring the advantage of easier updates from everywhere in the app but at the same time could lead to strange bugs if improperly used - state updated from everywhere and leaking everywhere.
I'd still think `Store`s should be scoped to smaller app components and a `..ScreenStore` should only live while that Screen lives and only be available there.

The only other `Store` that is already a singleton is `AppStore` which maybe more suitable for the usecase of syncing the state of various components based on which a CFR should be shown.
Alternatively we could use an `AbstractBinding` in which to observe `BrowserStore` updates.